### PR TITLE
Debug daily metrics GHA

### PR DESCRIPTION
**Pull request by e-belfer**
# Overview

What problem does this address?
Fixes problems with the archiving action created in #162.

What did you change in this PR?
- Update Github API token to have correct permissions
- Update GCS permissions configuration
- Fix typos and job name

# Testing

How did you make sure this worked? How can a reviewer verify this?
Run the `save-daily-metrics.yml` workflow and verify that it works as expected.

# To-do list

```[tasklist]
- [x] Review the PR yourself and call out any questions or issues you have
```

### CODE REVIEW

File: .github/workflows/save_daily_metrics.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         with:
           environment-file: environment.yml
           cache-environment: true
-          ondarc: |
```
Comment (jdangerx) [inline .github/workflows/save_daily_metrics.yml:25]: -_- my eyes completely glazed over this during the first review

File: .github/workflows/save_daily_metrics.yml [hunk 2]
```diff
@@ -41,9 +41,7 @@ jobs:
         continue-on-error: true
         uses: "google-github-actions/auth@v2"
         with:
-          workload_identity_provider: "projects/345950277072/locations/global/workloadIdentityPools/gh-actions-pool/providers/gh-actions-provider"
-          service_account: "pudl-usage-metrics-etl@catalyst-cooperative-pudl.iam.gserviceaccount.com"
-          create_credentials_file: true
+          credentials_json: "${{ secrets.GCP_USAGE_METRICS_ARCHIVER_KEY }}"
```
Comment (jdangerx) [inline .github/workflows/save_daily_metrics.yml:44]: Hmm, sure, but what was the issue with WIF?

File: .github/workflows/save_daily_metrics.yml [hunk 3]
```diff
@@ -53,7 +51,7 @@ jobs:
       - shell: bash -l {0}
         name: Save Github Metrics
         env:
-          API_TOKEN_GITHUB: ${{ secrets.API_TOKEN_GITHUB }}
+          API_TOKEN_GITHUB: ${{ secrets.USAGE_STATS_ACCESS_TOKEN }}
```
Comment (jdangerx) [inline .github/workflows/save_daily_metrics.yml:54]: Did you have to generate a new token or something?

**Review by jdangerx — APPROVED** [review]
:shipit: but also I have some questions about the various auth changes. Nothing blocking, just curious why they had to be done

File: .github/workflows/save_daily_metrics.yml (hunk 2 above)
Comment (e-belfer) [reply .github/workflows/save_daily_metrics.yml:44]: This was done to give this archiver write access and to keep the ETL account as having read-only access to the archived files.

File: .github/workflows/save_daily_metrics.yml (hunk 3 above)
Comment (e-belfer) [reply .github/workflows/save_daily_metrics.yml:54]: Yes, this was a reference to a secret that existed on the `business` repo. We can't authenticate within the GHA as accessing metrics requires admin level read-only access, so this required generating a new token.

File: .github/workflows/save_daily_metrics.yml (hunk 2 above)
Comment (jdangerx) [reply .github/workflows/save_daily_metrics.yml:44]: Makes sense! Do you have a new service account now? Is it in Terraform / if not, do you want help putting it in Terraform?

Comment (e-belfer) [reply .github/workflows/save_daily_metrics.yml:44]: Let me move this question into the living terraform PR (https://github.com/catalyst-cooperative/pudl/pull/3841), as @bendnorman set up the account and would know better.

**Pull request merged**